### ISSUE TITLE
Prevent getting blocked

### ISSUE THREAD

**Issue by marcelja** [marcelja/facebook-delete#51]
@AlexKalopsia mentioned in #50 that many users have to set a very high limit rate to not get blocked. It would be good if we can collect some ideas here on what we could do to not need such a high limit rate.

I'm wondering if it makes a difference that we have parallel requests enabled: https://github.com/marcelja/facebook-delete/blob/0590b9e55503be047f76d0812622c260abe82599/deleter.go#L20
Setting this to `1` could help to not get blocked.

Another thing that we could do: Not using the default go user agent. I already tried that but didn't get it working (#33).

**Comment by AlexKalopsia** [comment]
Yeah, that's a good point, I will try and test with 1 routine and report back. 

Using a different header also seems like a potentially good idea but haven't looked into it.

**Comment by aunefyren** [comment]
> collect some ideas here on what we could do to not need such a high limit rate.

I'm curious if the block is on an account basis or some IP detection. Could one theoretically change their IP after being blocked? If so, is it then possible to have nodes on different IP's collaborating on deleting different content without hitting the traffic limit?

**Comment by AKmatiAK** [comment]
ratelimit is progressive, so more you delete, more you have to extend deletion time. that's what I observed.

**Comment by AKmatiAK** [comment]
also, activity log on standard facebook seems to be out of mbasic ratelimit

**Comment by user583183** [comment]
i was quickly banned using default settings (100ms) with release 4.0 on ubuntu. i'll try 1500ms and see how that goes, tool seems to work but is being blocked

**Comment by AKmatiAK** [comment]
Interesting. When there is message "failed to delete" it actually deletes things even though ratelimit is imposed. Please someone confirm, as I observed this behaviorbut still I am not 100% sure.

**Comment by rajtslegr** [comment]
> Interesting. When there is message "failed to delete" it actually deletes things even though ratelimit is imposed. Please someone confirm, as I observed this behaviorbut still I am not 100% sure.

Yeah I think you are right.

**Comment by Esentries** [comment]
I'm not sure if this helps guys but I noticed that when I manually delete stuff in the activity log, 250 posts is the limit. I downloaded an autoscroller to make the process faster and consistently deleted 250 posts at a time, it didn't seem to block me.

**Comment by Tophsuki** [comment]
@Esentries how did you get it to only delete 250 posts? not sure how to enter commands once it loads. looks like I've already been blocked/banned though. Anyone know if the ban is temporary?

**Comment by autumnserenity** [comment]
> @AlexKalopsia mentioned in #50 that many users have to set a very high limit rate to not get blocked. It would be good if we can collect some ideas here on what we could do to not need such a high limit rate.
> 
> I'm wondering if it makes a difference that we have parallel requests enabled:
> 
> https://github.com/marcelja/facebook-delete/blob/0590b9e55503be047f76d0812622c260abe82599/deleter.go#L20
> 
> 
> Setting this to `1` could help to not get blocked.
> Another thing that we could do: Not using the default go user agent. I already tried that but didn't get it working (#33).

How can I modify the rate limit and how many posts it deletes? Also, would you  happen to know if the block/ban is temporary?

**Comment by devnoname120** [comment]
@Esentries mbasic.facebook.com, facebook.com, or a mobile app?
